table/transaction.go

```diff
@@ -142,6 +143,121 @@ func (t *Transaction) SetProperties(props iceberg.Properties) error {
 	return nil
 }
 
+type expireSnapshotsCfg struct {
+	minSnapshotsToKeep *int
+	maxSnapshotAgeMs   *int64
+}
+
+type ExpireSnapshotsOpt func(*expireSnapshotsCfg)
+
+func WithRetainLast(n int) ExpireSnapshotsOpt {
+	return func(cfg *expireSnapshotsCfg) {
+		cfg.minSnapshotsToKeep = &n
+	}
+}
+
+func WithOlderThan(t time.Duration) ExpireSnapshotsOpt {
+	return func(cfg *expireSnapshotsCfg) {
+		n := t.Milliseconds()
+		cfg.maxSnapshotAgeMs = &n
+	}
+}
+
+func (t *Transaction) ExpireSnapshots(opts ...ExpireSnapshotsOpt) error {
+	var (
+		cfg          expireSnapshotsCfg
+		updates      []Update
+		snapsToKeep  = make(map[int64]struct{})
+		refsToDelete = make(map[string]struct{})
+		nowMs        = time.Now().UnixMilli()
+	)
+
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	for refName, ref := range t.meta.refs {
+		if refName == MainBranch {
+			continue
+		}
+
+		snap, err := t.meta.SnapshotByID(ref.SnapshotID)
+		if err != nil {
+			return err
+		}
+
+		maxRefAgeMs := internal.Coalesce(ref.MaxRefAgeMs, cfg.maxSnapshotAgeMs)
+		if maxRefAgeMs == nil {
+			return errors.New("cannot find a valid value for maxRefAgeMs")
+		}
```

Comment:
Instead of needing an `internal.Coalesce` function we could use `cmp.Or(ref.MaxRefAgeMs, cfg.maxSnapshotAgeMs)` and achieve the same result without needing our own separate function.

Reply:
Indeed.